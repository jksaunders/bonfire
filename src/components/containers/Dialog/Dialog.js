import React from 'react';
import PropTypes from 'prop-types';
import MaterialDialog from '@material-ui/core/Dialog';

const propTypes = {
  children: PropTypes.node,
  onClose: PropTypes.func,
};

const defaultProps = {
  children: null,
  onClose: null,
};

const Dialog = ({ children, onClose, ...rest }) => (
  <MaterialDialog
    maxWidth={false}
    onBackdropClick={onClose == null ? () => {} : () => onClose()}
    {...rest}
  >
    {children}
  </MaterialDialog>
);

Dialog.propTypes = propTypes;
Dialog.defaultProps = defaultProps;
export default Dialog;
