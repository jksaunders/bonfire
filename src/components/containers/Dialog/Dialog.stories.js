import React, { useState } from 'react';

import { withKnobs } from '@storybook/addon-knobs';

import Button from '../../Button';
import Dialog from './Dialog';

export default {
  title: 'Containers|Dialog',
  component: Dialog,
  decorators: [withKnobs],
};

export const DialogStory = () => {
  const [open, setOpen] = useState(false);
  return (
    <div>
      <Button text="Open dialog" onClick={() => setOpen(!open)} />
      <Dialog onClose={() => setOpen(false)} open={open}>
        pls
      </Dialog>
    </div>
  );
};

DialogStory.story = {
  name: 'with text',
};
