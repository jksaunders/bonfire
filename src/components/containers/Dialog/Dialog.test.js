import React from 'react';
import { expectExists, expectSnapshot, render } from '../../../testing';
import Dialog from './Dialog';

const getDialog = (props = {}) =>
  render(<Dialog {...props}>test-dialog</Dialog>);

test('Dialog closed', () => {
  const { container } = getDialog({ open: false });
  expectSnapshot(container);
});

test('Dialog open', () => {
  const { getByText } = getDialog({ open: true });
  const component = getByText('test-dialog');
  expectExists(component);
  expectSnapshot(component);
});
