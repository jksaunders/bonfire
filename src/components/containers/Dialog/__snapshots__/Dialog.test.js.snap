// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog closed 1`] = `<div />`;

exports[`Dialog open 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthFalse MuiPaper-rounded"
  role="dialog"
>
  test-dialog
</div>
`;
